stdlib/email/src/main/java/org/ballerinalang/stdlib/email/server/EmailConsumer.java

```
@@ -0,0 +1,101 @@
+
+    private EmailListener emailListener;
+    private ObjectValue client;
+
+    /**
+     * Constructor for the EmailConsumer.
+     *
+     * @param emailProperties Map of property values
+     * @param listener Forwards the received emails to Ballerina code
+     */
+    public EmailConsumer(Map<String, Object> emailProperties, EmailListener listener) {
+        this.emailListener = listener;
+        String host = (String) emailProperties.get(EmailConstants.PROPS_HOST);
+        String username = (String) emailProperties.get(EmailConstants.PROPS_USERNAME);
+        String password = (String) emailProperties.get(EmailConstants.PROPS_PASSWORD);
+        String protocol = (String) emailProperties.get(EmailConstants.PROPS_PROTOCOL);
+        MapValue protocolConfig = (MapValue) emailProperties.get(EmailConstants.PROTOCOL_CONFIG);
+        if (protocol.equals(EmailConstants.IMAP)) {
+            client = BallerinaValues.createObjectValue(EmailConstants.EMAIL_PACKAGE_ID, EmailConstants.IMAP_CLIENT,
+                    host, username, password, (MapValue<Object, Object>) protocolConfig);
+            EmailAccessClient.initImapClientEndpoint(client, host, username, password,
+                    (MapValue<Object, Object>) protocolConfig);
+        } else if (protocol.equals(EmailConstants.POP)) {
+            client = BallerinaValues.createObjectValue(EmailConstants.EMAIL_PACKAGE_ID, EmailConstants.POP_CLIENT,
+                    host, username, password, (MapValue<Object, Object>) protocolConfig);
+            EmailAccessClient.initPopClientEndpoint(client, host, username, password,
+                    (MapValue<Object, Object>) protocolConfig);
+        } else {
+            String errorMsg = "Protocol should either be 'IMAP' or 'POP'.";
+            final EmailConnectorException e = new EmailConnectorException(errorMsg);
+            emailListener.onError(BallerinaErrors.createError(e));
```

Since we initializing Email Consumer in the email listener lifecycle, we need to return an error and allow the system to panic. The error in configuration can't be recovered

---

Done.